### ISSUE TITLE
No error is given for return in after visit statements

### ISSUE THREAD

**Issue by psybers** [boalang/compiler#44]
If you give a return inside a before visit statement, it properly provides an error message:

```
foo.boa: compilation failed: Encountered typecheck error at line 8, column 3. return statement not allowed inside visitors
            return;
            ^^^^^^^
```

However if you use a return inside an after visit statement, no error is given and the code actually compiles just fine.  This however is not the correct logic and breaks optimizations/transformations that rely on the fact a return never appears in visit statements.

**Comment by psybers** [comment]
https://github.com/boalang/compiler/blob/9432c2178672cbe4e2aaf2afd29d4193802ff2aa/src/java/boa/compiler/visitors/TypeCheckingVisitor.java#L753

**Comment by psybers** [comment]
Note that the logic for that check is very wrong, as written.  Not only does it allow returns in an after visit statement, but it will complain if a visit statement tries declaring a function with a return inside it, it will fail if there is a nested visitor:

```
visitor {
   before Method -> {
      visit(.., visitor {
         after Statement -> ..
      });
      return;
   }
}
```

and probably fails for other reasons too!
